### ISSUE TITLE
[Suggestion] Close final page after successful login

### ISSUE THREAD

**Issue by ConnorJC3** [int128/kubelogin#193]
This is a minor annoyance, but the final page in the browser-based login flow (the one that says "OK") will stay open after login is complete, polluting my browser's tabs (especially if it opened in the background and automatically logged me in). Maybe you could optionally serve JS on that page to automatically close it?

**Comment by int128** [comment]
Currently kubelogin serves the following HTML after login.

```html
<html><body>OK<script>window.close()</script></body></html>
```

It closes the tab only if there is no user interaction on login page. For example,

1. Log in to the Keycloak.
1. Run kubelogin.
1. Kubelogin opens the tab and redirects to the Keycloak.
1. Keycloak does not show the login page (already logged in at step 1).
1. Kubelogin closes the tab automatically.

This works with Keycloak and Okta. Not with Google IdP.

Let me know if you have better solution.

**Comment by ConnorJC3** [comment]
Ah, I was unaware that this was already implemented. Unfortunately, it does not work for me when using Keycloak with firefox - I get a "Scripts may not close windows that were not opened by script." error in the web console, and the page stays open.

**Comment by int128** [comment]
This is due to the limitation of `window.close()` as:

> https://developer.mozilla.org/en-US/docs/Web/API/Window/close
>
> This method can only be called on windows that were opened by a script using the Window.open() method. If the window was not opened by a script, an error similar to this one appears in the console: Scripts may not close windows that were not opened by script.

**Comment by sfxworks** [comment]
Possible alternatives:
- Alert or text that states "You may now close this window"
- Option to provide redirect instead of a close
- Option to customize the close page

**Comment by bartlettc22** [comment]
@sfxworks option 3 available with via #308

**Comment by int128** [comment]
Now the following features have been implemented. Please open a new feature request if needed.

> Alert or text that states "You may now close this window"
> Option to provide redirect instead of a close
